### PR TITLE
Adds 5 new Heretic spells. Rebalances some aspects of Heretics. Refactors some spells as well, and makes it so emote spells require free hands.

### PR DESCRIPTION
**Pull request by MrMelbert**
## About The Pull Request

- Adds 5 new heretic spells!
   - For Flesh: Flesh Surgery. This spell is a touch spell that can either be used to heal your minions or extract organs from mobs. 
   - For Void: Cone of Cold. This is a simple spell - it shoots out a cone, of cold, that freezes and damages people caught in it.
   - For Ash: Volcano Blast. This spell functions like Tesla Blast, but instead of electricity, it shoots out of a beam of fire that hurts to walk over. 
   - For Blade: Realignment. Think of this like "Fleshmend but for stuns / stamcrit". It rapidly regenerates stamina damage and reduces stuns, while making you a pacifist. It can also be cast in rapid succession, but this will increase the cooldown.
   - For Rust: Rust Construction. Point at a rusted tile, and a wall will be raised where it was instantly. This even damages people and throws them aside - Or, if on a multi-z map, can lift up. 

- Number of influences has increased.
  - 5 at 1 heretic
  - 9 at 2 
  - 12 at 3
  - 14 at 4
  - 15 at 5
  - 16 at 6, and so on

- Heretics are given a 5th sacrifice target, selected randomly. On average an additional sacrifice is needed for their objectives. 

- Being sacrificed grants you a permanent phobia of the supernatural. Phobia of the supernatural has been expanded to cover heretic items and mobs. 

- The equation for offhand damage of blade heretics was tweaked. Actual result unchanged, it's just more resilient to future changes now. 

- Touch spells were refactored a bit, and overall expanded to be easier to use

- Charged spells were added, and charged beam spells. Tesla blast uses this.

- Cone spells were refactored to be easier to setup.

- Jaunting will now hide your runechat when it triggers, to make it less easy to follow.

- Heretic Ghouls now take less stamina damage based on how low their health pool is.

- Emote based spells now require hands to be unblocked to be cast, like mime spells.
   - Yes this gets rid of handcuffed invisible walls... Not 100% on this, but I figured it's good for consistency? Open to discussion

## Why It's Good For The Game

A lotta feedback has passed through about heretic and it's time to address some of it 

- Problem: Not enough cool flash spells. Makes Focus not worth it. 
   - Solution: Adds some more spells to encourage focus use. 

- Problem: Sacrifice targets being too willing or not harmed enough
   - Solution: A permanent trauma. 

- Problem: Not enough ways to power up.
   - Solution: Adding more influences around, though I think there should be more variety in knowledge rituals as well. 

- Problem: Ash Passage sucks
   - Solution: Makes it a smidge better to stay hidden with it.

- Problem: Heretic Ghouls get one hit by batons
   - Solution: Stamina modifier should put them on par with unmodified humans.

## Changelog

:cl: Melbert
add: Added five new heretic spells, one for each path. They come after the Ritual of Knowledge.
add: Cone of Cold, for Void heretics. Shoots out a freezing chill in a cone which deal damage and freezes.
add: Flesh Surgery for Flesh heretics. A touch spell which can either heal minions or be used on mobs to extract organs without surgery. 
add: Volcano Blast for Ash heretics. A beam spell, like Tesla Blast, which fires out a beam of fire that bounces between people.
add: Realignment for Blade heretics. Fleshmend, but for stuns and stamina damage. Makes you a pacifist, but rapidly regenerates stamina. 
add: Rust Construction for Rust heretics. Places a wall of rust on the target rusted flooring. Can even be used to ascend z-levels!
balance: Nerfed the cooldown of Cleave slightly, buffed the cooldown of Lesser Cleave slightly. 
balance: Slightly more influences will spawn on the station per heretic.
balance: Heretics require an additional sacrifice on average for ascension, but are given a fifth sacrifice target (randomly selected).
balance: Being sacrificed by a heretic now gives you a permanent phobia of spooky things, including heretic mobs and items.
balance: Heretic ghouls now take reduced stamina damage, depending on how small their health pool is.
balance: Using Jaunts will conceal your runechat for their duration. 
balance: Spells which require emoting (Mime spells) require your hands not be blocked to use. 
refactor: Touch Spells were improved a bit. Added some new template spells - Charged spells, and Charged beam spells. 
fix: Fixes a runtime from losing heretic.
/:cl:

### CODE REVIEW

**Comment by private-tristan** [comment]
>  balance: Using Jaunts will conceal your runechat for their duration.

ash jaunt more useful now?!?! blessed

**Comment by Shadowflame909** [comment]
Mimes deserve it if it means heretic buffs. They can invisible wall their prison cell when the sec officer tries to release them!

Flesh Surgery should be useful now that thralls can't really disguise themselves anymore!

File: code/modules/antagonists/heretic/knowledge/flesh_lore.dm [hunk 1]
```diff
@@ -219,7 +217,19 @@
...
 /datum/heretic_knowledge/knowledge_ritual/flesh
+	next_knowledge = list(/datum/heretic_knowledge/spell/flesh_surgery)
+	route = PATH_FLESH
+
+/datum/heretic_knowledge/spell/flesh_surgery
+	name = "Knitting of Flesh"
+	desc = "Grants you the spell Knit Flesh. This spell allows you to remove organs from victims \
+		without requiring a lengthy surgery. This process is much longer if the target is not dead. \
+		This spell also allows you to heal your minions and summons, or restore failing organs to acceptable status."
+	gain_text = "But yhey were not out of my reach for long. With every step, the screams grew, until at last \
```
Comment (necromanceranne) [inline code/modules/antagonists/heretic/knowledge/flesh_lore.dm:228]:
```suggestion
	gain_text = "But they were not out of my reach for long. With every step, the screams grew, until at last \
```

**Comment by Redrover1760** [comment]
> > balance: Using Jaunts will conceal your runechat for their duration.
> 
> ash jaunt more useful now?!?! blessed

Wait, does this apply to all jaunts, including wizard jaunts?

That's awesome.

**Comment by CRITAWAKETS** [comment]
> * Jaunting will now hide your runechat when it triggers, to make it less easy to follow.

oh fuck i've gotta port this PR to two different servers now because admin events LOVE jaunting

**Comment by KittyNoodle** [comment]
Mime nerf 👍

**Comment by Livrah** [comment]
We got heretic buffs, but at what cost..
(mime nerf doesn't make sense, you can gesticulate when handcuffed)

**Comment by Jackal-boop** [comment]
Whats the temp of the void come blast?

**Comment by private-tristan** [comment]
> Whats the temp of the void come blast?

It appears to be the void chill status effect which does " 21 K (14 * TEMPERATURE_DAMAGE_COEFFICIENT)" that stacks with the mansus grasp version.

**Comment by Jackal-boop** [comment]
> > Whats the temp of the void come blast?
> 
> It appears to be the void chill status effect which does " 21 K (14 * TEMPERATURE_DAMAGE_COEFFICIENT)" that stacks with the mansus grasp version.

Oh so it doesn't effect local atmos?

**Comment by Fikou** [comment]
> We got heretic buffs, but at what cost.. (mime nerf doesn't make sense, you can gesticulate when handcuffed)

they dfeserve it

**Comment by Jackal-boop** [comment]
Melbert make it interact with atmos please 😃

**Comment by DustanHache** [comment]
While I like the ideas of this draft, i have one **glaring** balance issue to bring up: traumas will make the heretic even stronger overall, and also affect cult/revenant/wizards in turn by making them stronger in some ways. For this reason we should go the route of sleeper surgery and make the trauma deep rooted, so it is treatable but not trivial to remove without risk. If one or more heretics go on a rampage, the last thing you want is to give them random free stuns and blinds on previous victims.

**Comment by private-tristan** [comment]
can't cast void blast without wizard robes
image
image

**Comment by blackdav123** [comment]
> While I like the ideas of this draft, i have one **glaring** balance issue to bring up: traumas will make the heretic even stronger overall, and also affect cult/revenant/wizards in turn by making them stronger in some ways. For this reason we should go the route of sleeper surgery and make the trauma deep rooted, so it is treatable but not trivial to remove without risk. If one or more heretics go on a rampage, the last thing you want is to give them random free stuns and blinds on previous victims.

You died to the heretic. Getting a free aheal after being murdered by an antag is not something offered by any other antag. Heretics _should_ have an advantage over anyone they've already killed once.

**Comment by anclandor** [comment]
I imagine that the trauma will apply after they go through the trial, because if they got paralyzed before having to dodge hands  it would be kinda funny.

**Comment by DustanHache** [comment]
> > While I like the ideas of this draft, i have one **glaring** balance issue to bring up: traumas will make the heretic even stronger overall, and also affect cult/revenant/wizards in turn by making them stronger in some ways. For this reason we should go the route of sleeper surgery and make the trauma deep rooted, so it is treatable but not trivial to remove without risk. If one or more heretics go on a rampage, the last thing you want is to give them random free stuns and blinds on previous victims.
> 
> You died to the heretic. Getting a free aheal after being murdered by an antag is not something offered by any other antag. Heretics _should_ have an advantage over anyone they've already killed once.

If they want that they can do a proper roundremoval by mutilating the body so much it doesn’t get ahealed. There’s a check for that. 
It’s like looking at abduction objective and and saying “they should be permanently brain trauma’d after being abducted because its unfair to the tot” which is bullshit. The only difference between that objective and heretic is that heretic is FORCED to crit or kill their target before abducting.

**Comment by Jacquerel** [comment]
The phobia addition is, I assume, a response to a forum post stating that heretics should just kill people automatically.  
There's a not unreasonable assertion that as it stands, sacrifices don't "move the round forward" that much in that you "kill" someone but then they just pop right back, sometimes in better health than they were _before_ they were sacrificed. 
The thread also asserts that it is a problem for admins to enforce people not just immediately turning you in or hunting you down, despite what the text says about not remembering it.

The phobia now adds an inconvenience to that player and also makes them have great difficulty fighting the Heretic in the future, which seems like a reasonable solution to these problems.
Now something obviously fucked up is happening, and Heretics who are bringing their targets back to life don't need to worry overmuch about those people immediately coming back and trying to turn them horizontal, gradually converting the crew into a group of people who will have real trouble opposing them.

I'm inclined to like this because I think heretic sacrifices putting people in a funny hand room and then reviving them is more interesting than just killing them.

**Comment by Troxonater** [comment]
I disagree on the mime nerf, the amount of timing and luck it takes to get any actual value out of handcuffed invisible walls that isn't just a 30 second annoyance for sec (most officers take longer than that to even finish processing you) doesn't deserve a nerf.

**Comment by blackdav123** [comment]
are shadow shackles one of the feared objects? if yes it might make it impossible to break out of the cuffs once you are freed from mansus

**Comment by Redrover1760** [comment]
> are shadow shackles one of the feared objects? if yes it might make it impossible to break out of the cuffs once you are freed from mansus

Nope. The only restraints that is a feared object are cult bolas.

**Comment by arnociiroc** [comment]
thbis is just like avatar

**Comment by IndieanaJones** [comment]
Very good changes.

**Comment by ATH1909** [comment]
isn't rust supposed to be about causing metal to decay, not creating new metal?

**Comment by ATH1909** [comment]
image
booooooooooooooooo

trapping a sec officer in the same cell as you is, like, A-tier invisible wall comedy, bro

**Comment by ATH1909** [comment]
> Jaunting will now hide your runechat when it triggers, to make it less easy to follow.

would it be possible to instead make this make your runechat message stick in place as you jaunt away? or would that look jank as hell

**Comment by ATH1909** [comment]
how do heretic ghouls interact with stamina-healing chems? are aranesp-infused ghouls basically immune to being staminacrit?

**Comment by saprasam** [comment]
honestly man just bring back heretic sacrifices gibbing/disemboweling you, having a permanent phobia and a random brain trauma is an actively negative fun experience to deal with. for the rest of the round.
getting gibbed is more fun because at least i get to watch the heretic do evil things, but with a phobia i either get blinded, or hardstunned by the rifts in reality that are in the halls or in rooms

**Comment by YG-Dev** [comment]
I don't think that "Sacrifice targets are being too willing or not harmed enough". Permanent trauma is huge debuff, making it more painful than fleshy mess from abductor kidnapping. At least it should be cerebral with brain damage, so it still causes consequences

**Comment by EuclidianGeometer** [comment]
nah, permenant is good i think. it's weird, fucked up, and magical. it also stops that person from trying to take you down later.

**Comment by DustanHache** [comment]
> nah, permenant is good i think. it's weird, fucked up, and magical. it also stops that person from trying to take you down later.

Then lets standardize it at the very least: Make sleeper protocol give permanent conspiracy trauma instead of deep rooted. Make curse of madness for wizard also give solely permanent traumas.

**Comment by Oranggu** [comment]
> For Flesh: Flesh Surgery. This spell is a touch spell that can either be used to heal your minions or extract organs from mobs.

will this be able to pull organs out of living beings? hope not.

**Comment by Oranggu** [comment]
Happy this got merged 👍

**Pull request merged**